src/pages/auth/login.jsx

```diff
@@ -0,0 +1,70 @@
+import { getLoginDataFromServer } from "../../utilities/apis/apis";
+import "./auth.css"
+import { useReducer } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import { loginReducer } from "../../reducer/authReducer";
+import { useAuth } from "../../context/authContext";
+import { Sidebar } from "../../components/sidebar/sidebar";
+
+export const Login = () => {
+    const {authDispatch}= useAuth();
+    const navigate = useNavigate();
+
+    const [{email,password},loginDispatch] = useReducer(loginReducer, {email:"", password:""});
+    const testHandler=()=>[
+    loginDispatch({type:"SET_EMAIL",payload:"adarshbalika@gmail.com"}),
+    loginDispatch({type:"SET_PASSWORD",payload:"adarshBalika123"})
+    ]
+    const submitHandler= async (e, email, password)=>{
+        e.preventDefault();
+        try{
+        const loginResponse = await getLoginDataFromServer(email,password);
+        console.log(loginResponse)
+        localStorage.setItem("encodedToken", loginResponse.data.encodedToken)
+        localStorage.setItem('userData', JSON.stringify(loginResponse.data.foundUser));
+        authDispatch({ type: "USER_LOGIN" })
+        authDispatch({ type: "USER_TOKEN", payload: loginResponse.data.encodedToken })
+        authDispatch({ type: "USER_DATA", payload: loginResponse.data.foundUser })
+        navigate("/")
```

Comment:
You can navigate programmatically to the previous page that user was on before user was redirected to login page

Reply:
Sure i will add that to the code